zarr/core.py

```diff
@@ -180,6 +181,7 @@ def __init__(
 
         self._store = store
         self._chunk_store = chunk_store
+        self._transformed_chunk_store = None
```

Comment:
If we have a transform, do we need to maintain a link to the base store?
Could have ._chunk_store point to the transformed one, and not have to have the `if is None` checks below.

Reply:
Also, we should take note that only the data is getting transformed, not the metadata